operator/internal/central/extensions/reconcile_tls.go

```diff
@@ -142,12 +152,44 @@ func (r *createCentralTLSExtensionRun) validateAndConsumeCentralTLSData(fileMap
 	if err := r.ca.CheckProperties(); err != nil {
 		return errors.Wrap(err, "loaded service CA certificate is invalid")
 	}
+	if err := r.checkCertificateTimeValidity(r.ca.Certificate()); err != nil {
+		return errors.Wrap(err, "primary CA is not valid at the present time")
+	}
+
+	if centralCARotationEnabled.BooleanSetting() {
+		rotationAction, err := r.handleCARotation(fileMap)
+		if err != nil {
+			return err
+		}
+		r.caRotationAction = rotationAction
+		if r.caRotationAction != carotation.NoAction {
+			return errors.New("CA rotation action needed")
+		}
+	}
+
 	if err := r.validateServiceTLSData(storage.ServiceType_CENTRAL_SERVICE, "", fileMap); err != nil {
 		return errors.Wrap(err, "verifying existing central service TLS certificate failed")
 	}
 	return nil
 }
 
+func (r *createCentralTLSExtensionRun) handleCARotation(fileMap types.SecretDataMap) (carotation.Action, error) {
```

Comment:
"Handle CA rotation" sounds like it does the rotation, while this just seems to load, check and determine action, so how about:
```suggestion
func (r *createCentralTLSExtensionRun) checkSecondaryCA(fileMap types.SecretDataMap) (carotation.Action, error) {
```